data-entry/invoice-editor/invoice-editor-flow/src/main/java/com/vaadin/invoice/editor/InvoiceEditor.java

```diff
@@ -142,21 +143,36 @@ public InvoiceEditor() {
 
         // Grid Pro
         GridPro<Invoice> grid = new GridPro<>();
-        grid.setItems(createItems());
+        List<Invoice> invoiceList = createItems();
+        grid.setItems(invoiceList);
+
+        addCardTransactionBtn.addClickListener(e -> {
+            invoiceList.add(0, new Invoice("", 0, Currency.EUR, 0, 0, Category.PERSONAL, false, 0));
+            grid.getDataProvider().refreshAll();
+        });
+
         grid.addThemeVariants(GridVariant.LUMO_COLUMN_BORDERS,
                 GridVariant.LUMO_COMPACT);
 
-        grid.addEditColumn(Invoice::getProduct).text((item,
-                newValue) -> displayNotification("Product", item, newValue))
+        grid.addEditColumn(Invoice::getProduct, "product").text((item,
+                newValue) -> {
+            item.setProduct(newValue);
+            displayNotification("Product", item, newValue);
+        })
                 .setHeader("Product");
-        grid.addEditColumn(Invoice::getDescription)
-                .text((item, newValue) -> displayNotification("Description",
-                        item, newValue))
+        grid.addEditColumn(Invoice::getDescription, "description")
+                .text((item, newValue) -> {
+                    item.setDescription(newValue);
+                    displayNotification("Description",
+                            item, newValue);
+                })
                 .setHeader("Description").setWidth("250px");
-        grid.addEditColumn(Invoice::getPrice)
-                .text((item, newValue) -> displayNotification("Price", item,
-                        newValue))
-                .setHeader("Price").setTextAlign(ColumnTextAlign.END);
+        grid.addEditColumn(Invoice::getPrice, "price")
+                .text((item, newValue) -> {
+                    item.setPrice(Float.valueOf(newValue));
```

Comment:
The field allows to input any text which throws a `NumberFormatException` here if the string can't be converted to number.

Reply:
Surrounded with `try` now.